File: 7kyu/(7 kyu) Last/(7 kyu) Last.go
package kata;

import "reflect";

func Last(xs interface{}) interface{} {
  r := reflect.ValueOf(xs);
  
  if r.Kind() == reflect.String {
    var s = xs.(string);
    return s[len(s)-1:]
  }
  
  return r.Index(r.Len()-1).Interface();
}

File: 7kyu/(7 kyu) Last/(7 kyu) Last.py
# #1
# def last(*args):
#     last = args[-1]
#     try:
#         return last[-1]
#     except TypeError:
#         return last
#2
def last(*args): 
    try:
        return args[-1][-1]
    except:
        return args[-1]
